sanity/lib/utils.ts

```diff
@@ -1,10 +1,35 @@
-import client from './sanity';
-import getImage from './getImage';
+import { createClient } from 'next-sanity';
+
+import { config } from './config';
+
+export const client = createClient(config);
+
+//* HOME PAGE *\\
+export const fetchSpecificSkills = async <T extends String>(
+  skill: T | T[]
+) => {
+  const fetchedSkills = await client.fetch<any[]>(
+    `*[_type == "skills" && name ${
+      Array.isArray(skill) && !(skill instanceof String)
+        ? `in [${skill.map((skill) => `"${skill}"`).join(', ')}]`
+        : `match ${skill.trim()}`
+    }]{
+      ...,
+      "ref": icon.asset._ref
+    }`
+  );
+
+  if (!fetchedSkills) {
+    console.error('Error fetching those skill', fetchedSkills);
+    return null;
+  }
+
+  return fetchedSkills;
+};
```

Comment:
```suggestion
export const fetchSpecificSkills = async (skill: string[]) => {
  const fetchedSkills = await client.fetch(
    `*[_type == "skills" && name in ${JSON.stringify(skill)}]{
      name,
      bgColor,
      "imgURL": icon.asset->url
    }`
  );

  if (!fetchedSkills) {
    console.error('Error fetching those skill', fetchedSkills);
    return [];
  }

  return fetchedSkills;
};
```

Let's simplify this. We don't need to generically type the function or it's return- we know we're sending an array of strings. The data must always be an array or strings because that's the only type of data we'll be able to match with a name in your cms.

Let's not map over each item and join them, we can just stringify the array.

We can also get just the things we want (including the assetURL that we can use to display the image), instead of spreading to get everything with this section:
```js
      name,
      bgColor,
      "imgURL": icon.asset->url
 ```
 
 Just like what we talked about before, everything inside of the brackets describes the shape of the data we want returned.
 
 But honestly I don't think you should be doing this. The idea of a CMS is that we want to be able to edit and control what is displayed on the site without rebuilding the entire site or messing with code. The way you've set it up: we're relying on a hardcoded array to tell us which skills to display. At that point, you may as well add a "url" to each name in the array and then do away with the CMS.
 
 Why don't we add a boolean to the skills?
 
 ```js
 fields: [
    {
      name: 'name',
      title: 'Name',
      type: 'string',
    },
    {
      name: 'isMain',
      title: 'Is Main Skill',
      type: 'boolean',
    },
    {
      name: 'bgColor',
      title: 'BgColor',
      type: 'string',
    },
    { ...etc...
```
    

  And then search based on this boolean?
  
  ```js
    const fetchedSkills = await client.fetch(
    `*[_type == "skills" && isMain == true]{
      name,
      bgColor,
      "imgURL": icon.asset->url
    }`
  );
  ```
  
  
  That way we can control which ones end up on our main page and toggle them on or off without having to rebuild the site.
  
  
![image](https://github.com/jacastanon01/portfolio/assets/4108484/86a8b669-3026-4888-9a88-9ac04ea79b1f)